public/pages/workflows/new_workflow/utils.ts

```diff
@@ -138,68 +148,103 @@ export function fetchEmptyUIConfig(): WorkflowConfig {
   };
 }
 
-export function fetchSemanticSearchMetadata(): UIState {
+export function fetchSemanticSearchMetadata(version: string): UIState {
+  const isPreV219 = semver.lt(version, MINIMUM_FULL_SUPPORTED_VERSION);
   let baseState = fetchEmptyMetadata();
   baseState.type = WORKFLOW_TYPE.SEMANTIC_SEARCH;
-  baseState.config.ingest.enrich.processors = [new MLIngestProcessor().toObj()];
+
+  baseState.config.ingest.enrich.processors = isPreV219
+    ? [new TextEmbeddingIngestProcessor().toObj()]
+    : [new MLIngestProcessor().toObj()];
+
   baseState.config.ingest.index.name.value = generateId('knn_index', 6);
   baseState.config.ingest.index.settings.value = customStringify({
     [`index.knn`]: true,
   });
-  baseState.config.search.request.value = customStringify(TERM_QUERY_TEXT);
-  baseState.config.search.enrichRequest.processors = [
-    injectQueryTemplateInProcessor(
-      new MLSearchRequestProcessor().toObj(),
-      KNN_QUERY
-    ),
-  ];
+
+  baseState.config.search.request.value = customStringify(
+    isPreV219 ? SEMANTIC_SEARCH_QUERY_NEURAL : KNN_QUERY
+  );
```

Comment:
`KNN_QUERY` should be `TERM_QUERY_TEXT` to keep the original logic consistent.

Reply:
addressed on line 166